appium/webdriver/common/touch_action.py

```diff
@@ -24,16 +24,24 @@
 # pylint: disable=no-self-use
 
 import copy
+from typing import TYPE_CHECKING, Dict, List, TypeVar, Union
 
 from appium.webdriver.mobilecommand import MobileCommand as Command
 
+if TYPE_CHECKING:
+    from appium.webdriver.webelement import WebElement as WebElement
+    from appium.webdriver.webdriver import WebDriver as WebDriver
+
+T = TypeVar('T', bound='TouchAction')
+
 
 class TouchAction(object):
-    def __init__(self, driver=None):
+
+    def __init__(self, driver: 'WebDriver' = None):
         self._driver = driver
-        self._actions = []
+        self._actions: List = []
 
-    def tap(self, element=None, x=None, y=None, count=1):
+    def tap(self: T, element: 'WebElement' = None, x: int = None, y: int = None, count: int = 1) -> T:
```

Comment:
Do we need to set the type of `self` argument?

Reply:
Yes, in my understanding.

https://github.com/appium/python-client/pull/482#discussion_r365553785